tensorflow_model_optimization/python/core/common/keras/compression/algorithms/weight_clustering.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Weight clustering algorithm using tfmot compression api."""
```

Comment:
Would elaborate on this comment and clarify that this example differs from the current Keras weight clustering API, both in terms of functionality (e.g. ability to cluster a single layer) and in terms of method of application (this does decompression during inference instead of before inference with gzip/undoing gzip).